v1/src/main/java/com/google/cloud/teleport/bigtable/BeamRowToBigtableFn.java

```diff
@@ -148,41 +180,116 @@ public void processElement(
+        SetCell cell = createCell(defaultColumnFamily.get(), field.getName(), value, timestamp);
+        mutationsToAdd.addAll(Arrays.asList(Mutation.newBuilder().setSetCell(cell).build()));
       } else if (type.isCollectionType()) {
-        mutationsToAdd = createCollectionMutations(row, field);
+        mutationsToAdd.addAll(createCollectionMutations(row, field, timestamp));
       } else if (type.isMapType()) {
-        mutationsToAdd = createMapMutations(row, field);
+        mutationsToAdd.addAll(createMapMutations(row, field, timestamp));
       } else {
         throw new UnsupportedOperationException(
             "Mapper does not support type:" + field.getType().getTypeName().toString());
       }
+    }
 
-      for (Mutation mutationToAdd : mutationsToAdd) {
-        mutations.add(mutationToAdd);
-        cellsProcessed++;
-        if (this.splitLargeRows && cellsProcessed % maxMutationsPerRequest == 0) {
-          // Send a MutateRow request when we have accumulated max mutations per request.
-          out.output(KV.of(rowkey, mutations.build()));
-          mutations = ImmutableList.builder();
-        }
+    // DoFn return value.
+    ImmutableList.Builder<Mutation> mutations = ImmutableList.builder();
+    // Iterate over constructed mutations and build them. Split them up if the split large rows flag
+    // is on.
+    for (Mutation mutationToAdd : mutationsToAdd) {
+      mutations.add(mutationToAdd);
+      cellsProcessed++;
+      if (this.splitLargeRows && cellsProcessed % maxMutationsPerRequest == 0) {
+        // Send a MutateRow request when we have accumulated max mutations per request.
+        out.output(KV.of(rowkey, mutations.build()));
+        mutations = ImmutableList.builder();
       }
     }
-
     // Flush any remaining mutations.
     ImmutableList remainingMutations = mutations.build();
     if (!remainingMutations.isEmpty()) {
       out.output(KV.of(rowkey, remainingMutations));
     }
   }
 
+  private HashMap<String, Long> getColumnWritetimes(Row row, List<Field> nonKeyColumns) {
+    // Writetime values will be joined to the other mutations as timestamps.
+    HashMap<String, Long> columnTimestamps = new HashMap<>();
+    Pattern writetimeColumn = Pattern.compile(String.format("%s\\((.*)\\)", WRITETIME_COLUMN));
+    for (Field field : nonKeyColumns) {
+      if (field.getName().contains(WRITETIME_COLUMN)) {
+        // Get column name of writetime column, e.g. "writetime(column1)" will be the writetime for
+        // "column1".
+        Matcher matcher = writetimeColumn.matcher(field.getName());
+        matcher.matches();
+        String columnName = matcher.group(1);
+
+        // Get timestamp and normalize as necessary.
+        Long rawTimestamp = (Long) row.getValue(field.getName());
+        Long normalizedTimestamp = normalizeToBigtableTimestamp(rawTimestamp);
+
+        columnTimestamps.put(columnName, normalizedTimestamp);
+      }
```

Comment:
we are silently ignoring the timestamp column here, should we be raising the flag here? There are 2 cases where this is not good:

1. User's column names have special write time token which should be avoided, how do we inform them that its not a good practice? 
2. There is some bug in the regex or column name has special character that we are not handling well. We should raise a warning here, instead of silently ignoring the timestmap.

Reply:
Added a warning